config/knexfile.ts

```diff
@@ -1,58 +1,197 @@
+      doc: 'How long knex should wait when acquiring a connection before throwing a timeout error',
+      format: 'Number',
+      default: 30000
+    },
+    createTimeoutMillis: {
+      doc: 'How long knex should wait when establishing a new connection before throwing a timeout error',
+      format: 'Number',
+      default: 30000
+    },
+    destroyTimeoutMillis: {
+      doc: 'How long knex should wait when destroying new connection before throwing a timeout error',
+      format: 'Number',
+      default: 5000
+    },
+    idleTimeoutMillis: {
+      doc: 'How long a connection must sit idle in the pool and not be checked out before it is automatically closed',
+      format: 'Number',
+      default: 30000
+    },
+    reapIntervalMillis: {
+      doc: 'How often to check for idle resources to destroy',
+      format: 'Number',
+      default: 1000
+    },
+    createRetryIntervalMillis: {
+      doc: 'How long knex should wait before trying to create a connection again',
+      format: 'Number',
+      default: 200
     }
   },
-  test: {
-    client: 'sqlite3',
-    connection: ':memory:',
-    useNullAsDefault: true,
-    migrations: {
-      directory: migrationsDirectory,
-      tableName: 'auth-service',
-      loadExtensions: ['.ts']
-    },
-    seeds: {
-      directory: seedsDirectory,
-      loadExtensions: ['.ts']
+  migrations: {
+    directory: {
+      doc: 'Migration directory',
+      format: String,
+      default: migrationsDirectory
+    },
+    tableName: {
+      doc: 'Migration table name',
+      format: String,
+      default: 'auth-service'
+    },
+    stub: {
+      doc: 'Where the stubs for migration are located',
+      format: String,
+      default: `${migrationsDirectory}/migration.template`
+    },
+    loadExtensions: {
+      doc: 'Array of extensions to load',
+      format: 'Array',
+      default: ['.ts']
     }
   },
-  production: {
-    /* For now, the production environment is the same as the development environment. */
-    client: 'mysql',
-    version: '5.5',
-    connection: config.get('DATABASE').connection,
-    pool: config.get('DATABASE').pool,
-    migrations: {
-      directory: migrationsDirectory,
-      tableName: 'auth-service',
-      stub: `${migrationsDirectory}/migration.template`,
-      loadExtensions: ['.ts']
-    },
-    seeds: {
-      directory: seedsDirectory,
-      loadExtensions: ['.ts']
+  seeds: {
+    directory: {
+      doc: 'Seeds directory',
+      format: String,
+      default: seedsDirectory
+    },
+    loadExtensions: {
+      doc: 'Array of extensions to load',
+      format: 'Array',
+      default: ['.ts']
     }
   }
-}
+})
+
+const env = process.env.NODE_ENV ?? 'production'
+const dbConfigFile = `${__dirname}/${env}_db.json`
+ConvictDatabaseConfig.loadFile(dbConfigFile)
+ConvictDatabaseConfig.validate({ allowed: 'strict' })
+
+const Config: DatabaseConfig = ConvictDatabaseConfig.getProperties()
+
+// Inject directory paths here
```

Comment:
@eoln I removed the directory specification from config files themselves because the path resolution causes problems without using __dirname and resolving the absolute paths and I can't do that in the config file itself since its json and not script.